### DISCUSSION TITLE
Port iNav to EagleTree Vector hardware???

### DISCUSSION THREAD

**Discussion by csurf** [iNavFlight/inav#6563] [Ideas]
Just wanted to open up a discussion about the possibility to port iNav to the ET Vector, now that the company has gone out of business and has decided to no longer support the hardware. 

I realize that the hardware specs are very, very different from the standard iNav-compatible platform, but I figured it might be interesting to at least have the conversation & investigate the possibility of creating a port for the PIC32MX / ET Vector. The fact that it does color graphics seems like a very cool feature that iNav could possibly adopt.

I'd be willing to try to help with testing & development. I'd have to get an extra pair of "virgin' PIC32MX & flash chips and replace the ones on my board (otherwise, I'd lose the bootloader & firmware of the original Vector firmware). Others interested in testing would either have to use their own vector or try to find a used/bricked one for testing.

If anyone else is interested, please let me know. I can provide pictures & more info.

Here's a breakdown of the hardware & what I've discovered so far...

hardware:
- MCU: Microchip PIC32MX695F512L
- FLASH: Micron M25PX16 16mb serial flash
- MPU: Invensense MPU-6050
- BARO: TE Connectivity MS5607-02BA03

on-board OSD components:
- EL 1883ISZ analog video sync separator
- MAX4617 analog multiplexer
- LMV324 op amp
- unknown 28-pin SOP co-processor chip
- NTSC & PAL XTAL's (3.5795MHz & 4.4336MHz)

The co-processor chip is most likely another type of PIC based on the footprint / output pin layout. Unfortunately, the chip has been black-topped; the ID markings have been scraped off, so it's impossible to know for a fact. I wasn't able to read anything from it using a PICKIT 3 clone, so it's either an OTP chip, has had the programming pins disabled, or I simply did something wrong when trying to read the chip. 

The co-processor PIC is responsible for doing the color encoding for the Vector's color OSD. The main PIC32MX MCU communicates with the co-processor (via i2c), and it outputs some analog waveforms on a few pins at the FSC frequency. These waveforms are then passed through the op-amp and multiplexer chips in order to mix the color waveforms into the output video signal. I'm guessing that the main processor probably sends the color encoding for each video line via i2c, and then the co-processor outputs some type of RGB signaling, or some other means of doing the phase-shifting for the color encoding. Unfortunately, I'm not knowledgeable enough to understand this part of the analog circuitry, so someone else would need to help with decipering how this works. I can provide some logic dumps of the i2c comms if needed.

The PIC32MX apparently has a (custom) bootloader that handles built-in USB comms. USB is used for talking to the configuration software and doing firmware upgrades. However, USB is done via a custom protocol, and the PIC32MX presents itself as an HID device. In other words, it's currently not setup as a standard UART / COM port via USB. 

The PIC32MX is also responsible for generating the base video signal & OSD graphics (in black&white). This is apparent due to the fact that the OSD video output will still work even if there's no valid input signal. It also handles the optional audio output. The 1883 chip is used for all of the video sync'ing for the co-processor & PIC32MX.

I haven't looked at the GPS, mag, and current sensor communications, but i'm guessing it's done via i2c or serial.

**Comment by DzikuVx** [comment]
To be honest, I don't see it happening. Different architecture.
